rcp/org.testeditor.rcp4/src/org/testeditor/rcp4/dialogs/NetworkConnectionSettingDialog.java

```diff
@@ -359,46 +377,15 @@ protected void okPressed() {
 	 * @return if it is possible to reach internet resources.
 	 */
 	public boolean isInternetAvailable(boolean updateSessings) {
+		if (ignoreConnectionState) {
+			return true;
+		}
 		if (workOffline) {
 			System.setProperty(WORKOFFLINE, Boolean.toString(workOffline));
 			return true;
 		}
 		if (System.getProperty(PROXY_HOST) == null | updateSessings) {
-			if (proxyHostSetting != null && proxyHostSetting.length() > 0) {
-				IProxyService proxyService = context.get(IProxyService.class);
-				proxyService.setProxiesEnabled(true);
-				proxyService.setSystemProxiesEnabled(true);
-				IProxyData proxyData = proxyService.getProxyData(IProxyData.HTTP_PROXY_TYPE);
-				IProxyData httpsProxyData = proxyService.getProxyData(IProxyData.HTTPS_PROXY_TYPE);
-				proxyData.setHost(proxyHostSetting);
-				httpsProxyData.setHost(proxyHostSetting);
-				if (proxyPortSetting.length() > 0) {
-					int port = Integer.parseInt(proxyPortSetting);
-					proxyData.setPort(port);
-					httpsProxyData.setPort(port);
-				}
-				proxyData.setUserid(proxyUserSetting);
-				proxyData.setPassword(proxyPwdSetting);
-				httpsProxyData.setUserid(proxyUserSetting);
-				httpsProxyData.setPassword(proxyPwdSetting);
-				try {
-					if (noProxyHostsSetting.length() > 0) {
-						proxyService.setNonProxiedHosts(noProxyHostsSetting.split(","));
-					}
-				} catch (CoreException e) {
-					e.printStackTrace();
-				}
-				System.setProperty(PROXY_HOST, proxyHostSetting);
-				System.setProperty(PROXY_HTTPS_HOST, proxyHostSetting);
-				System.setProperty(PROXY_PORT, proxyPortSetting);
-				System.setProperty(PROXY_HTTPS_PORT, proxyPortSetting);
-				System.setProperty(PROXY_USER, proxyUserSetting);
-				System.setProperty(PROXY_PWD, proxyPwdSetting);
-				System.setProperty(NON_PROXY_HOSTS, noProxyHostsSetting);
-				if (pathToMavenSettingsFileSettings != null && pathToMavenSettingsFileSettings.length() > 0) {
-					System.setProperty(PATH_TO_MAVENSETTINGS, pathToMavenSettingsFileSettings);
-				}
-			}
+			updateSystemSettings();
```

Comment:
+1 for extracting this into a method. To make even cleaner, I would prefer not calling this from `isInternetAvailable`. From the name I would expect that the method only checks a state and doesn't update any settings.